static/scripts/render-transaction.ts

```diff
@@ -93,21 +94,10 @@ const insertTableData = async (table: Element): Promise<Element> => {
 };
 
 async function renderEnsName(element: Element, address: string, tokenView: boolean = false): Promise<void> {
-  // const provider = new ethers.providers.Web3Provider(window.ethereum);
-  // const ens = await provider.lookupAddress(address);
-  const ensResolve = await fetch(`https://ens.cirip.io/${address}`);
+  const provider = new ethers.providers.JsonRpcProvider("https://rpc-pay.ubq.fi/v1/mainnet");
   let href: string = "";
   try {
-    const resolved = await ensResolve.json();
-    let ensName;
-    if (resolved.reverseRecord) {
-      ensName = resolved.reverseRecord;
-    } else if (resolved.domains.length) {
-      const domain = resolved.domains.shift();
-      if (domain) {
-        ensName = domain;
-      }
-    }
```

Comment:
I need to test this further. This was an interesting fallback I made for when people improperly configured their ENS.

Reply:
Currently stuck on resolving these complications. Their docs are really confusing for me! https://docs.ens.domains/dapp-developer-guide/resolving-names